composables/useNft.ts

```diff
@@ -52,9 +58,7 @@ async function getRmrk2Resources(nft: NFTWithMetadata) {
 }
 
 async function getProcessMetadata(nft: NFTWithMetadata) {
-  const metadata = (await processSingleMetadata(
-    nft.metadata
-  )) as NFTWithMetadata
+  const metadata = await processSingleMetadata(nft.metadata)
```

Comment:
can we force it by adding `as NFTWithMetadata` or maybe something else in here?

ts warning on this side 
![image](https://github.com/kodadot/nft-gallery/assets/734428/56bff114-1d41-49c8-b418-d600af4a7b00)